scripts/globals/garrison.lua

```diff
@@ -513,129 +172,246 @@ local spawnNPCs = function(npc)
-            for _, entityId in pairs(zoneData.npcs) do
-                -- TODO: Don't use GetMobByID here
-                -- Keep a table with { bool: alive/dead, mob object } entries
-                local entity = GetMobByID(entityId)
-                if entity and entity:isAlive() then
-                    allNPCsDead = false
-                end
-            end
+        [xi.garrison.state.GRANT_LOOT] = function()
+            debugLog("State: Grant Loot")
+            debugPrintToPlayers(zoneData, "Mission success")
+
+            xi.garrison.handleLootRolls(xi.garrison.loot[zoneData.levelCap], zoneData.players)
+            zoneData.state = xi.garrison.state.ENDED
+        end,
+
+        [xi.garrison.state.ENDED] = function()
+            debugLog("State: Ended")
 
-            local allPlayersDead = true
             for _, entityId in pairs(zoneData.players) do
                 local entity = GetPlayerByID(entityId)
-                -- TODO: Only check valid players
-                if entity and entity:isAlive() then
-                    allPlayersDead = false
+                if entity ~= nil then
+                    entity:delStatusEffect(xi.effect.LEVEL_RESTRICTION)
                 end
             end
 
-            local allMobsDead = true
-            for _, entityId in pairs(zoneData.mobs) do
-                -- TODO: Don't use GetMobByID here
-                -- Keep a table with { bool: alive/dead, mob object } entries
-                local entity = GetMobByID(entityId)
-                if entity and entity:isAlive() then
-                    allMobsDead = false
-                end
+            for _, entityId in pairs(zoneData.npcs) do
+                DespawnMob(entityId, zone)
             end
 
-            if
-                allNPCsDead or
-                allPlayersDead or
-                allMobsDead
-            then
-                -- TODO: Only check valid players
-                for _, entityId in pairs(zoneData.players) do
-                    local entity = GetPlayerByID(entityId)
-                    entity:delStatusEffect(xi.effect.LEVEL_RESTRICTION)
-                end
-
-                for _, entityId in pairs(zoneData.npcs) do
-                    DespawnMob(entityId)
-                end
-
-                for _, entityId in pairs(zoneData.mobs) do
-                    DespawnMob(entityId)
-                end
-
-                zoneData.state = xi.garrison.state.ENDED
+            for _, entityId in pairs(zoneData.mobs) do
+                DespawnMob(entityId, zone)
             end
-        end,
 
-        [xi.garrison.state.ENDED] = function()
-            -- print("State: Ended")
             zoneData.continue = false
         end,
     }
 
-    -- TODO: Check to see if current wave is complete
-
-    -- TODO: Check to if all waves are completed
-
     if zoneData.continue then
-        npc:timer(2000, function(npcArg)
+        npc:timer(1000, function(npcArg)
```

Comment:
2s tick seemed a bit too slow, esp when it drives the respawn timers that are already at 15 secs, so pushed it down to 1s